src/modules/cpu/kernel/ricap.hpp

```diff
@@ -0,0 +1,1629 @@
+                    rpp_simd_store(rpp_store48_u8pln3_to_u8pln3, dstPtrTempR, dstPtrTempG, dstPtrTempB, p); // simd stores
+                    srcPtrTemp4 += vectorIncrement;
+                    dstPtrTempR += vectorIncrementPerChannel;
+                    dstPtrTempG += vectorIncrementPerChannel;
+                    dstPtrTempB += vectorIncrementPerChannel;
+                }
+                for (; vectorLoopCount4 < bufferLength[3]; vectorLoopCount4 += 3)
+                {
+                    *dstPtrTempR = srcPtrTemp4[0];
+                    *dstPtrTempG = srcPtrTemp4[1];
+                    *dstPtrTempB = srcPtrTemp4[2];
+                    srcPtrTemp4 += 3;
+                    dstPtrTempR++;
+                    dstPtrTempG++;
+                    dstPtrTempB++;
+                }
+
+                srcPtrRow3 += srcDescPtr->strides.hStride;
+                srcPtrRow4 += srcDescPtr->strides.hStride;
+                dstPtrRowR += dstDescPtr->strides.hStride;
+                dstPtrRowG += dstDescPtr->strides.hStride;
+                dstPtrRowB += dstDescPtr->strides.hStride;
+            }
+        }
+
+        // Ricap with fused output-layout toggle (NCHW -> NHWC)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NCHW) && (dstDescPtr->layout == RpptLayout::NHWC))
+        {
+            Rpp8u *srcPtrRowR1, *srcPtrRowG1, *srcPtrRowB1, *srcPtrRowR2, *srcPtrRowG2, *srcPtrRowB2;
+            Rpp8u *srcPtrRowR3, *srcPtrRowG3, *srcPtrRowB3, *srcPtrRowR4, *srcPtrRowG4, *srcPtrRowB4, *dstPtrRow;
+            srcPtrRowR1 = srcPtrChannel[0];
+            srcPtrRowG1 = srcPtrRowR1 + srcDescPtr->strides.cStride;
+            srcPtrRowB1 = srcPtrRowG1 + srcDescPtr->strides.cStride;
+            srcPtrRowR2 = srcPtrChannel[1];
+            srcPtrRowG2 = srcPtrRowR2 + srcDescPtr->strides.cStride;
+            srcPtrRowB2 = srcPtrRowG2 + srcDescPtr->strides.cStride;
+            srcPtrRowR3 = srcPtrChannel[2];
+            srcPtrRowG3 = srcPtrRowR3 + srcDescPtr->strides.cStride;
+            srcPtrRowB3 = srcPtrRowG3 + srcDescPtr->strides.cStride;
+            srcPtrRowR4 = srcPtrChannel[3];
+            srcPtrRowG4 = srcPtrRowR4 + srcDescPtr->strides.cStride;
+            srcPtrRowB4 = srcPtrRowG4 + srcDescPtr->strides.cStride;
+            dstPtrRow = dstPtrChannel;
+
+            for (int i = 0; i < roi[0].xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTempR1, *srcPtrTempG1, *srcPtrTempB1, *srcPtrTempR2, *srcPtrTempG2, *srcPtrTempB2, *dstPtrTemp;
+                srcPtrTempR1 = srcPtrRowR1;
+                srcPtrTempG1 = srcPtrRowG1;
+                srcPtrTempB1 = srcPtrRowB1;
+                srcPtrTempR2 = srcPtrRowR2;
+                srcPtrTempG2 = srcPtrRowG2;
+                srcPtrTempB2 = srcPtrRowB2;
+
+                dstPtrTemp = dstPtrRow;
+
+                int vectorLoopCount1 = 0;
```

Comment:
For each vector loop possibly add a comment saying // Top-Left Quadrant // Top-Right Quadrant // Bottom-Left Quadrant // Bottom-Right Quadrant

Reply:
Sure